### PR TITLE
[19.0][MIG] sale_delivery_split_date: Migration to 19.0

### PR DESCRIPTION
**Pull request by Tatider**
* An update of tests was required because in Odoo 19.0, the database by default doesn't have Demo Data.

Depends on:

- [x] [sale_order_line_date migration](https://github.com/OCA/sale-workflow/pull/3946)
- [x] [sale_procurement_group_by_line migration](https://github.com/OCA/sale-workflow/pull/3947)

### CODE REVIEW

File: sale_delivery_split_date/models/sale_order_line.py [hunk 1]
```diff
@@ -10,33 +10,39 @@
...
+    def _prepare_procurement_values(self):
+        vals = super()._prepare_procurement_values()
+        line_com_date = self.commitment_date
+        if line_com_date:
             comm_date = self._get_security_lead_time_commitment_date()
-            vals.update({"date_planned": comm_date})
+            vals.update(
+                date_planned=comm_date,
+                date_deadline=line_com_date,
+            )
```
Comment (yankinmax) [inline sale_delivery_split_date/models/sale_order_line.py:45]: non-blocking, just suggestion
```
if line_com_date := self.commitment_date:
    comm_date = self._get_security_lead_time_commitment_date()
    vals.update(
        date_planned=comm_date,
        date_deadline=line_com_date,
    )
```

File: sale_delivery_split_date/static/description/index.html [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
 <meta name="generator" content="Docutils: https://docutils.sourceforge.io/" />
-<title>Sale Delivery Split Date</title>
```
Comment (yankinmax) [inline sale_delivery_split_date/static/description/index.html]: Is this change generated by `pre-commit`? Strange change to `<title>README.rst</title>`

**Review by yankinmax — APPROVED** [review]
pre-approve considering a question

File: sale_delivery_split_date/static/description/index.html (hunk 2 above)
Comment (Tatider) [reply sale_delivery_split_date/static/description/index.html]: Yes, it is precommit doing the change...

File: sale_delivery_split_date/models/sale_order_line.py (hunk 1 above)
Comment (Tatider) [reply sale_delivery_split_date/models/sale_order_line.py:45]: Thank you for the suggestion, applied.

**Comment by Tatider** [comment]
One of the dependencies to be renamed, waiting for the final name to be defined - "https://github.com/OCA/sale-workflow/pull/3947"

**Comment by Tatider** [comment]
> One of the dependencies to be renamed, waiting for the final name to be defined - "#3947"

Updated

File: sale_delivery_split_date/models/sale_order_line.py [hunk 3]
```diff
@@ -10,33 +10,39 @@
...
-    def _prepare_procurement_group_vals(self):
-        vals = super()._prepare_procurement_group_vals()
-        if self._get_procurement_group_key()[0] == 24:
+    def _prepare_reference_vals(self):
+        vals = super()._prepare_reference_vals()
+        if self._get_stock_reference_key()[0] == 24:
             if self.commitment_date:
                 comm_date = self._get_security_lead_time_commitment_date()
-                vals["name"] += "/" + format_date(self.env, comm_date.date())
+                vals["name"] = (
```
Comment (ivantodorovich) [inline sale_delivery_split_date/models/sale_order_line.py:18]: Maybe it's just me, but I found it was more readable before.
Why this change? it doesn't seem like an useful diff

**Review by ivantodorovich — APPROVED** [review]
small comment, non blocking

File: sale_delivery_split_date/models/sale_order_line.py (hunk 3 above)
Comment (Tatider) [reply sale_delivery_split_date/models/sale_order_line.py:18]: To my understanding, usage of "+=" for strings is considered a bad practice, bacause it might lead to unxpected errors when one of arguments is not a string, f.e. "False" as it might be with Sale Line Name

**Comment by yaracchi** [comment]
Are the 2 failing checks blocking or this can be used?

**Comment by Tatider** [comment]
> Are the 2 failing checks blocking or this can be used?

It could be used. Warnings were about test coverage and the old status of dependencies.
Anyway, all is resolved now.

**Comment by yankinmax** [comment]
Hello @rousseldenis can this PR be merged now? All suggestions are taken into account.

**Comment by imlopes** [comment]
Hello @OCA/crm-sales-marketing-maintainers 

Could you check and merge this PR ? 
It seems all comments are fixed and it's ready.

Thanks a lot

**Review by alexey-pelykh — APPROVED** [review]
Thorough migration from 18.0 to 19.0. The module correctly adapts to the renamed dependency (`sale_procurement_group_by_line` -> `sale_stock_reference_by_line`) and the new `stock.reference` model API. Both dependency PRs (#3946, #3947) are merged. CI is fully green.

**Key migration changes reviewed:**

1. **Dependency update** -- `sale_procurement_group_by_line` renamed to `sale_stock_reference_by_line` with comment `# ex. sale_procurement_group_by_line` for traceability. Good.

2. **API adaptation** -- `_prepare_procurement_group_vals()` correctly renamed to `_prepare_reference_vals()` to match the new base module API. The string concatenation refactored from `vals["name"] +=` to explicit `vals.get("name", "") + ...` -- reasonable defensive approach as discussed in prior review comments.

3. **`_prepare_procurement_values`** -- Method signature correctly drops the `group_id` parameter to match the 19.0 base. The walrus operator usage (`if line_com_date := self.commitment_date:`) is clean and was already reviewed/approved. The addition of `date_deadline=line_com_date` alongside `date_planned` is a welcome improvement to propagate the commitment date as the move deadline.

4. **Tests** -- Properly adapted from `TransactionCase` to `TestSaleStockCommon` to avoid demo data dependency (which Odoo 19.0 no longer provides by default). All `procurement_group_id` references updated to `stock_reference_id`. New `test_wo_a_date` test covers the no-commitment-date edge case. The `test_check_same_dates` and security lead time tests now also verify `date_deadline` on stock moves -- good additional coverage.

5. **Views** -- `stock_picking.xml` unchanged from 18.0 logic, makes `scheduled_date` visible in tree and searchable. Standard OCA pattern.

No blocking issues found. LGTM.

**Review by paulius-sladkevicius — APPROVED** [review]
LGTM

**Comment by yankinmax** [comment]
Hello @OCA/crm-sales-marketing-maintainers

Could you check and merge this PR ?

**Comment by yankinmax** [comment]
Hello @rousseldenis can you pls merge this PR?

**Review by rousseldenis — APPROVED** [review]
